lib/utilities.js

```diff
@@ -96,9 +96,35 @@ function setStyleProp(style, props) {
  */
 var PRESERVE_CUSTOM_ATTRIBUTES = React.version.split('.')[0] >= 16;
 
+// Taken from
+// https://github.com/facebook/react/blob/cae635054e17a6f107a39d328649137b83f25972/packages/react-dom/src/client/validateDOMNesting.js#L213
+var elementsWithNoTextChildren = new Set([
+  'tr',
+  'tbody',
+  'thead',
+  'tfoot',
+  'colgroup',
+  'table',
+  'head',
+  'html',
+  'frameset',
+  '#document'
+]);
+
+/**
+ * Returns True if the the given node can contain text nodes
+ * @param {DomElement} node
+ * @returns {boolean}
+ */
+function canTextBeChildOfNode(node) {
+  return !elementsWithNoTextChildren.has(node.name);
```

Comment:
Can you check node validity here?

```js
return node && node.parent && !elementsWithNoTextChildren.has(node.name);
```

Reply:
I don't really know if that makes sense.

This function only checks if the passed in node can contain a text node or not. eg
`<table>` -> No, it can't contain a text
`<div>` -> Yes, it can't contain a text

If I do the check you've suggested, then that changes the meaning of the function. i.e "is the passed in node a valid child of its parent"

Reply:
Makes sense, let's keep it as-is